pkg/sampler/sampler.go

```diff
@@ -33,15 +35,13 @@ func (t *Target) SetHash() {
 
 type Sample struct {
 	StatusCode      int
-	T1              time.Time
-	T2              time.Time
-	ResponseHeaders map[string]string
-}
-
-// Latency returns the amount of milliseconds between T1
-// and T2 (start and finish).
-func (s Sample) Latency() float64 {
-	return s.T2.Sub(s.T1).Seconds() * 1000
+	TimeStart       time.Time
+	TimeToConnect   time.Time
+	TimeToFirstByte time.Time
+	TimeEnd         time.Time
```

Comment:
I've been chewing on this for a few days, trying to come up with something reasonable.  I'm not married to the current implementation but figure I can at least expand on my starting place.

I was originally inspired by libcurl's implementation.  In that world, your time to connect and time to first byte metrics are both deltas from the original start time (not deltas from the previous phase).  I found the metrics to be great but the choice of delta to be inflexible, but I assume the author and some subset of the libcurl community is happy with that choice.  So for this implementation, I figure I'd just use raw timestamps and let users calculate the deltas as they see fit.  Some people may want a delta from the beginning like libcurl, some might want a delta from the previous phase.